nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_chat_dataset.py

```diff
@@ -157,14 +196,21 @@ def _build_samples_mapping(self):
         assert hasattr(self.tokenizer, "vocab"), "tokenizer should have vocab property, not supported"
         assert '<extra_id_0>' in self.tokenizer.vocab, "<extra_id_0> not in the tokenizer vocab. not supported"
         assert '<extra_id_1>' in self.tokenizer.vocab, "<extra_id_1> not in the tokenizer vocab. not supported"
+        # calcuilate <extra_id_2> id value
+        if '<extra_id_2>' in self.tokenizer.vocab:
+            ids_1 = self.tokenizer.text_to_ids('<extra_id_1><extra_id_2>')
```

Comment:
is this a typo?

Reply:
Not sure why we can't get the text_to_ids('<extra_id_2>') directly. This looks pretty hacky

Reply:
this is to handle the sentencepiece tokenizer which adds a special token in front. A hacky way I agree.

Reply:
Btw this check is super slow since `self.tokenizer.vocab` is a list not a dictionary/set. It is actually faster to do `len(self.tokenizer.text_to_ids('<extra_id_2>')) == 1`.

Reply:
Hmm, nit but isn't it just gonna be `self.extra_id_2_id = self.tokenizer.text_to_ids('<extra_id_2>')[-1]`